Plugins/Events/Events/ResourceEvents.cpp

```diff
@@ -0,0 +1,161 @@
+#include "Events/ResourceEvents.hpp"
+#include "API/CExoBase.hpp"
+#include "API/CExoAliasList.hpp"
+#include "API/CNWSModule.hpp"
+#include "API/CExoResMan.hpp"
+#include "API/Functions.hpp"
+#include "API/Constants.hpp"
+#include "Events.hpp"
+#include "Utils.hpp"
+
+#include <sys/inotify.h>
+#include <unistd.h>
+#include <poll.h>
+#include <climits>
+#include <fcntl.h>
+
+namespace Core {
+    extern bool g_CoreShuttingDown;
+}
+
+namespace Events {
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+using namespace NWNXLib::API::Constants;
+
+#define EVENT_SIZE          ( sizeof (struct inotify_event) )
+#define EVENT_BUF_LEN       ( 128 * ( EVENT_SIZE + NAME_MAX + 1) )
+
+static bool processEvents;
+static int selfPipeFds[2];
+std::unique_ptr<std::thread> ResourceEvents::m_pollThread;
+
+ResourceEvents::ResourceEvents(Services::TasksProxy* tasks)
+{
+    Events::InitOnFirstSubscribe("NWNX_ON_RESOURCE_.*", [tasks]()
+    {
+        int ret = pipe2(selfPipeFds, O_NONBLOCK);
+
+        if (ret == -1)
+        {
+            LOG_ERROR("Failed to initialize self-pipe with error: %s", std::strerror(errno));
+            return;
+        }
+
+        int pipeReadFd = selfPipeFds[0];
+        CExoString exoPath = Globals::ExoBase()->m_pcExoAliasList->GetAliasPath("NWNX");
```

Comment:
Should it also watch DEVELOPMENT:?

Reply:
Shouldn't be too hard to add, personally I only use NWNX: for my stuff. If you think it's useful I can add a watch for it.

Reply:
Added DEVELOPMENT: after all!